Userland/Libraries/LibCore/Socket.cpp

```diff
@@ -200,14 +215,22 @@ void PosixSocketHelper::setup_notifier()
 ErrorOr<NonnullOwnPtr<TCPSocket>> TCPSocket::connect(ByteString const& host, u16 port)
 {
     auto ip_address = TRY(resolve_host(host, SocketType::Stream));
-    return connect(SocketAddress { ip_address, port });
+
+    return ip_address.visit(
+            [port](IPv6Address ipv6_address) { return connect(SocketAddress { ipv6_address, port }); },
+            [port](IPv4Address ipv4_address) { return connect(SocketAddress { ipv4_address, port }); }
```

Comment:
You should be able to put this in one visitor:
```suggestion
            [port](auto address) { return connect(SocketAddress { address, port }); },
```